app/services/record_declarations/base.rb

```diff
@@ -110,5 +120,33 @@ def validate_provider!
       # TODO: Remove the nil? check and fix the test setup so that they build the school cohort, partnership and give us back the actual lead_provider.
       raise ActionController::ParameterMissing, I18n.t(:invalid_participant) unless actual_lead_provider.nil? || lead_provider_from_token == actual_lead_provider
     end
+
+    def parsed_date
+      Time.zone.parse(declaration_date)
+    end
+
+    def validate_schedule!
+      schedule = user_profile.schedule
+
+      unless schedule
+        raise ActionController::ParameterMissing, I18n.t(:schedule_missing)
+      end
+
+      existing_declarations = user_profile.participant_declarations
+
+      if existing_declarations.count >= schedule.milestones.count
+        raise ActionController::ParameterMissing, I18n.t(:too_many_declarations)
+      end
+
+      next_milestone = schedule.milestones[existing_declarations.count]
+      parsed_date = self.parsed_date
```

Comment:
i think you can delete this line. i don't see the point of assigning the method return to a local variable of the same name?

Reply:
Ah, I think this is an artifact of an old refactor - when I was parsing inline